From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: femboyPig <60929459+femboyPig@users.noreply.github.com>
Date: Thu, 30 Mar 2023 09:37:46 +0300
Subject: [PATCH] Stackable items


diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index fc481f26368f4d823379921573a05ac5b91f39e6..08c78905b75f76d7ffc5d99b084b03f342202c47 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -570,6 +570,12 @@ public final class ItemStack {
     }
 
     public int getMaxStackSize() {
+        // YellowKnife start - stackable items
+        if (this.tag != null && this.tag.contains("YellowKnife.MaxStackSize")) {
+            byte maxStackSize = this.tag.getByte("YellowKnife.MaxStackSize");
+            if (maxStackSize > 0 && maxStackSize <= 64) return maxStackSize;
+        }
+        // YellowKnife end
         return this.getItem().getMaxStackSize();
     }
 
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index 6556d7ab09826bb5a99f11385eddc26b67e44d68..49d1f41e4010255b1ab1a98fc4cefca39d32fcd9 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -188,7 +188,7 @@ public final class CraftItemStack extends ItemStack {
 
     @Override
     public int getMaxStackSize() {
-        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getItem().getMaxStackSize();
+        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getMaxStackSize(); // YellowKnife - stackable items
     }
 
     // Paper start
